### PR TITLE
Link to documentation page for "folder marker missing"

### LINKED ISSUE #5207: vague arror message: "folder marker missing"
The error "folder marker missing" is kind of vague.
Looking around, it means that the " .stfolder " file is missing.

So, why doesn't the error message say so?

> the file ".stfolder" is missing from the folder

### PR DESCRIPTION
**Pull request by bill-mcgonigle**
Fixes #5207.  The docs site can be updated as needed with additional information as desired.
Unfortunately no HTML anchor to link to on the docs site, but that could come later.

### Purpose

Describe the purpose of this change. If there is an existing issue that is
resolved by this pull request, ensure that the commit subject is on the form
`Some short description (fixes #1234)` where 1234 is the issue number.

### Testing

Describe what testing has been done, and how the reviewer can test the change
if new tests are not included.

### Screenshots

If this is a GUI change, include screenshots of the change. If not, please
feel free to just delete this section.

### Documentation

If this is a user visible change (including API and protocol changes), add a link here
to the corresponding pull request on https://github.com/syncthing/docs or describe
the documentation changes necessary.

## Authorship

Your name and email will be added automatically to the AUTHORS file
based on the commit metadata.

### CODE REVIEW

File: lib/config/folderconfiguration.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 var (
 	ErrPathNotDirectory = errors.New("folder path not a directory")
 	ErrPathMissing      = errors.New("folder path missing")
-	ErrMarkerMissing    = errors.New("folder marker missing (this indicates potential data loss, search docs/forum to get information about how to proceed)")
+	ErrMarkerMissing    = errors.New("folder marker missing (consult the 'Error' section on this page: https://docs.syncthing.net/intro/gui.html?highlight=.stfolder )")
```
Comment (tomasz1986) [inline lib/config/folderconfiguration.go:28]:
```suggestion
	ErrMarkerMissing    = errors.New("folder marker missing (consult the 'Error' section on this page: https://docs.syncthing.net/intro/gui.html)")
```

Adding a link to the Docs is useful, but please don't remove the original message though. What if the Docs website isn't working when the problem happens? The user will be left with nothing.

**Comment by bill-mcgonigle** [comment]
There is documented severe dataloss from people consulting bad advice on the forum (that's how this PR came to be), so I would not feel comfortable putting my name on a commit that led them to that end.  If the docs site is down (how often?  how often is the forum down?) I would say archive.org would be the right approach.

**Comment by tomasz1986** [comment]
The general idea is to at least have a very brief explanation directly in the code, without having to rely on any external source. There may be cases, where people use Syncthing in a strictly offline/intranet environment with Internet access restricted, or in countries, where all of those sites are blocked due to censorship. They should still be able to know what is going on without visiting a specific site.

Linking to the Docs as a supplementary resource is fine, of course, although to be honest, there're already links to both the Docs and the forum directly at the bottom of the Web GUI, hence the error message only mentions the places without any URLs hard-coded there.

File: lib/config/folderconfiguration.go (hunk 1 above)
Comment (imsodin) [reply lib/config/folderconfiguration.go:28]: I think the highlighting is deliberate - at least it does point to the right place.

**Comment by imsodin** [comment]
I'd personally rather have a dedicated entry somehwere for this (FAQ, ...). The example in the GUI is pretty brief and it seems circumstantial that it is there.

I'd keep the bit about "this indicates potential data loss" - otherwise there's no indication that this is a potentially pretty grave situation.

**Comment by st-review** [comment]
:robot: beep boop

I'm going to close this pull request as it has been idle for more than 90 days.

This is not a rejection, merely a removal from the list of active pull requests that is periodically reviewed by humans. The pull request can be reopened when there is new activity, and merged once any remaining issues are resolved.

To permanently exempt this PR from bot nagging, add the `slow-pr` label.

**Pull request closed without merging**